.github/workflows/core-ci.yml

```
@@ -52,92 +52,79 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          # Fetch enough history to validate commits
           fetch-depth: 0
-          # Check out the PR head commit directly (avoid synthetic merge ref)
           ref: ${{ github.event.pull_request.head.sha }}
 
-      - name: Setup git configuration
+      - name: Validate PR metadata
         run: |
```

actionlint (1.7.9)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)